pkg/bundle/sync.go

```diff
@@ -113,6 +122,75 @@ func (b *bundle) secretBundle(ctx context.Context, ref *trustapi.SourceObjectKey
 	return string(data), nil
 }
 
+func (b *bundle) httpsBundle(ctx context.Context, ref *trustapi.SourceHTTPS) (string, error) {
+	// Only use CAs from the SourceHTTPS inline bundle:
+	cas := x509.NewCertPool()
+	var block *pem.Block
+	certBytes := []byte(ref.CABundle)
+	for {
+		block, certBytes = pem.Decode(certBytes)
+		if block == nil {
+			break
+		}
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			// shouldn't happen as we have passed the validating webhook at this point
+			return "", err
+		}
+		cas.AddCert(cert)
+	}
+	// Make HTTP client that uses the root CAs from the pool
+	c := &http.Client{
+		// Thought: make this configurable?
+		Timeout: 10 * time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: cas,
+			},
+		},
+	}
+	// Construct request
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, ref.URL, nil)
+	if err != nil {
+		return "", fmt.Errorf("internal error constructing http request: %w", err)
+	}
+	// If we have cached data already, add etag and date to the request
+	b.etagCache.RLock()
+	if cachedData, found := b.etagCache.cache[ref.URL]; found {
+		req.Header.Add("If-None-Match", cachedData.etag)
+		req.Header.Add("If-Modified-Since", cachedData.date)
+	}
+	b.etagCache.RUnlock()
+	resp, err := c.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("couldn't make http request: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotModified {
+		b.etagCache.RLock()
+		bundleData := b.etagCache.cache[ref.URL].data
+		b.etagCache.RUnlock()
+		return string(bundleData), nil
+	}
+	body, err := io.ReadAll(resp.Body)
```

Comment:
suggestion: This feels too trusting of the server! The server could have returned anything except StatusNotModified here and we'd cache the response body and use it, even if it was a 404 or 500 or any error.

Would it be useful here if we checked that the response is 2xx and checking that the response looks something like a trust bundle? E.g. if we can't parse at least one PEM encoded certificate in the response, we shouldn't cache the response and use it.